libbeat/common/worker.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	"sync"
+)
+
+type WorkerSignal struct {
+	Done    chan struct{}
```

Comment:
Tried it and got ```invalid operation: close(ws.Done) (cannot close receive-only channel)```.

Solution would be to put a fixed number of ```struct{}``` in the read-only channel when creating it with ```NewWorkerSignal```. But that means we know how many Go routines have to be shutdown at ```WorkerSignal``` initialization. 
Even then, we can't be sure that each Go routine consume only once from the channel.